lib/auth/auth_with_roles.go

```diff
@@ -2503,6 +2503,18 @@ func (a *ServerWithRoles) CreateAccessRequestV2(ctx context.Context, req types.A
 		}
 	}
 
+	if !req.GetState().IsPending() {
+		if authz.IsCurrentUser(a.context, req.GetUser()) {
+			return nil, trace.AccessDenied("cannot create access request for self in non-pending state")
+		}
+
+		if len(a.context.Identity.GetIdentity().ActiveRequests) != 0 {
+			return nil, trace.AccessDenied("cannot create access requests in non-pending state while using an access request")
+		}
+
+		log.Warnf("Use of resource-level access request 'create' permission by user %q to create non-pending access request for user %q. Creation of non-pending requests will be deprecated in future version of teleport. Consider migrating to the review-based workflow instead.", a.context.User.GetName(), req.GetUser())
```

Comment:
> Consider migrating to the review-based workflow instead.

Do we really want to say this? One one hand we want enterprise users to fix their stuff, on the other hand we are directing OSS users to something they cannot do? Can we log this only if we're running teleport enterprise? For exmaple if we have SAML or OIDC in the modules.

Reply:
Good point, this is worded poorly.  I'll rephrase it to something like "approval workflow" so that it can refer equally to something like `tctl request approve`/`SetAccessRequestState` or `tsh request review --approve`/`SubmitAccessReview` equally.